tutorials/physics/using_character_body_2d.rst

```diff
@@ -258,34 +254,20 @@ Attach a script to the CharacterBody2D and add the following code:
 
     using Godot;
 
-    public partial class KBExample : CharacterBody2D
+    public partial class CBExample : CharacterBody2D
     {
-        public int Speed = 250;
-        private Vector2 _velocity = new Vector2();
+        public int Speed = 300;
 
         public void GetInput()
         {
-            // Detect up/down/left/right keystate and only move when pressed
-            _velocity = new Vector2();
-
-            if (Input.IsActionPressed("ui_right"))
-                _velocity.x += 1;
-
-            if (Input.IsActionPressed("ui_left"))
-                _velocity.x -= 1;
-
-            if (Input.IsActionPressed("ui_down"))
-                _velocity.y += 1;
-
-            if (Input.IsActionPressed("ui_up"))
-                _velocity.y -= 1;
-            _velocity = _velocity.Normalized() * Speed;
+            Vector2 inputDir = Input.GetVector("ui_left", "ui_right", "ui_up", "ui_down");
+            Velocity = inputDir * Speed;
         }
 
-        public override void _PhysicsProcess(float delta)
+        public override void _PhysicsProcess(double delta)
         {
             GetInput();
-            MoveAndCollide(_velocity * delta);
+            MoveAndCollide(Velocity * delta);
```

Comment:
Since Vector2 uses `float` and delta is a `double`, it needs to be converted to `float` so it can be multiplied.

```suggestion
            MoveAndCollide(Velocity * (float)delta);
```

Reply:
Dang it - I remembered to change `delta` to a `double`...